### ISSUE TITLE
Doc links to methods on primitives aren't resolved

### ISSUE THREAD

**Issue by SludgePhD** [rust-lang/rust-analyzer#20473]
**rust-analyzer version**: 0.3.2555-standalone

**rustc version**: 1.89.0

**editor or extension**: VSCode

**relevant settings**: none

**code snippet to reproduce**:
```rust
/// [`slice::sort`] [`slice::copy_within`]
fn f() {}
```

Neither of the links are resolved by rust-analyzer. rustdoc does not complain about them and links them to the right method.

If `use std::slice;` is in scope, `slice::sort` will be incorrectly resolved to the internal `sort` module.

It seems like rustdoc treats these methods specially, since you cannot actually refer to `slice::sort` in Rust code like that.
